ccm_web/client/src/hooks/useGlobals.ts

```diff
@@ -4,32 +4,44 @@ import usePromise from './usePromise'
 import * as api from '../api'
 import { Globals } from '../models/models'
 
-/* Hook for fetching global data and checking whether user is authenticated. */
-function useGlobals (ltiKey: string | undefined): [
+/*
+Hook for fetching global data, checking whether user is authenticated, and
+requesting that the backend set the CSRF token cookie
+*/
+function useGlobals (): [
   Globals | undefined,
   boolean | undefined,
   boolean,
+  Error | undefined,
   Error | undefined
 ] {
   const [globals, setGlobals] = useState(undefined as Globals | undefined)
 
-  const getGlobals = async (): Promise<Globals> => await api.getGlobals(ltiKey)
+  const getGlobals = async (): Promise<Globals> => await api.getGlobals()
   const [doGetGlobals, getGlobalsLoading, getGlobalsError] = usePromise(
     getGlobals,
     (value: Globals) => setGlobals(value)
   )
+  const setCSRFTokenCookie = async (): Promise<void> => await api.setCSRFTokenCookie()
+  const [doSetCSRFTokenCookie, setCSRFTokenCookieLoading, setCSRFTokenCookieError] = usePromise(setCSRFTokenCookie)
+
   useEffect(() => {
     void doGetGlobals()
   }, [])
 
+  useEffect(() => {
+    void doSetCSRFTokenCookie()
+  }, [])
+
   let isAuthenticated
-  if (ltiKey !== undefined && globals !== undefined) {
+  if (globals !== undefined) {
     isAuthenticated = true
   } else if (getGlobalsError !== undefined) {
     isAuthenticated = false
   }
 
-  return [globals, isAuthenticated, getGlobalsLoading, getGlobalsError]
+  const loading = getGlobalsLoading || setCSRFTokenCookieLoading
+  return [globals, isAuthenticated, loading, getGlobalsError, setCSRFTokenCookieError]
```

Comment:
Does it seem reasonable to handle the CSRF cookie setting in `useGlobals`? Or should it be somewhere else?